src/blueapi/service/runner.py

```diff
@@ -138,13 +138,13 @@ def _rpc(
     function_name: str,
     expected_type: type[T] | None,
     *args: Any,
-    **kwargs: Any,
+    **P: Any,
 ) -> T:
     mod = import_module(module_name)
-    func: Callable[P, T] = _validate_function(
+    func: Callable[..., T] = _validate_function(
         mod.__dict__.get(function_name, None), function_name
     )
-    value = func(*args, **kwargs)
+    value = func(*args, **P)
```

Comment:
This does not look right. P is a ParamSpec, here you're not just naming the dictionary constructed from un-splatting any kwargs as `P`.

Reply:
nevermind, it's ok now